src/panels/config/integrations/integration-panels/ssdp/dialog-ssdp-raw-data.ts

```diff
@@ -0,0 +1,67 @@
+import { LitElement, html, nothing, css } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import type { TemplateResult } from "lit";
+import { fireEvent } from "../../../../../common/dom/fire_event";
+import type { HassDialog } from "../../../../../dialogs/make-dialog-manager";
+import { createCloseHeading } from "../../../../../components/ha-dialog";
+import type { HomeAssistant } from "../../../../../types";
+import "../../../../../components/ha-code-editor";
+
+export interface SSDPRawDataDialogParams {
+  key: string;
+  data: Record<string, unknown>;
+}
+
+@customElement("dialog-ssdp-raw-data")
+class DialogSSDPRawData extends LitElement implements HassDialog {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _params?: SSDPRawDataDialogParams;
+
+  public async showDialog(params: SSDPRawDataDialogParams): Promise<void> {
+    this._params = params;
+  }
+
+  public closeDialog(): boolean {
+    this._params = undefined;
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+    return true;
+  }
+
+  protected render(): TemplateResult | typeof nothing {
+    if (!this._params) {
+      return nothing;
+    }
+
+    return html`
+      <ha-dialog
+        open
+        @closed=${this.closeDialog}
+        .heading=${createCloseHeading(
+          this.hass,
+          `${this.hass.localize("ui.panel.config.ssdp.raw_data_title")}: ${this._params.key}`
+        )}
+      >
+        <ha-code-editor
+          mode="yaml"
+          .value=${JSON.stringify(this._params.data, null, 2)}
```

Comment:
You can use `import { dump } from "js-yaml"` to show it as YAML since `<ha-code-editor>` is configured for YAML. Unless there is a reason to prefer JSON
```suggestion
          .value=${dump(this._params.data)}
```